plugins/deployment_manager/Deploy.ts

```diff
@@ -20,7 +21,7 @@ export interface DeployOpts {
   name?: string; // name for aliasing
   overwrite?: boolean; // should we overwrite existing contract link
   connect?: Signer; // signer for the returned contract
-  verify?: boolean; // verify contract on etherscan
+  verificationStrategy?: VerificationStrategy; // strategy for verifying contracts on etherscan
```

Comment:
I guess maybe my question now is, why do we ever need an eager strategy?

Reply:
Good question. Mainly just preserving the status quo, but also maybe if want to avoid writing/reading to cache for various use cases (smaller deploys, scenarios).

Reply:
Still not convinced we need anything but lazy, but this all looks very clean 🧼 I think we should merge, but also, if the complexity of supporting multiple verification strategies leads to any problems or difficulties, I think we should consider only supporting lazy

Reply:
Definitely, shouldn't be too hard to only support lazy if we ever want to in the future